library/objective-c/EnvoyHTTPStreamImpl.m

```diff
@@ -8,7 +8,8 @@
 #pragma mark - Utility types and functions
 
 typedef struct {
-  EnvoyHTTPCallbacks *callbacks;
+  __unsafe_unretained EnvoyHTTPCallbacks *callbacks;
+  __unsafe_unretained EnvoyHTTPStreamImpl *stream;
```

Comment:
Do you mind putting a comment here that explains why we store this ref back to the stream, and how the lifetime of the stream is tied to the context rather than the callback?

Reply:
I had it in the PR description but will add here too